[docs] reflects updates to cloudflare: streams, env variables (#5694)
withastro/astro: packages/integrations/cloudflare/README.md

Depending on your adapter mode (advanced = worker, directory = pages), the runtime object will look a little different due to differences in the Cloudflare API.

## Environment Variables

> As Cloudflare Pages Functions [provides environment variables per request](https://developers.cloudflare.com/workers/platform/environment-variables/), you can only access private environment variables when a request has happened. Usually, this means moving environment variable access inside a function.

Author: ```suggestion
See Cloudflare's documentation for [working with environment variables](https://developers.cloudflare.com/pages/platform/functions/bindings/#environment-variables).
```
Not sure whether the advice is still relevant, since their documentation and some basic default usage on their end has changed since these docs were written. So, doing this until someone PRs something different!  :sweat_smile:
